src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerDriver.cs

```diff
@@ -114,11 +125,25 @@ private void Initialize(Compilation comp, AnalyzerExecutor analyzerExecutor, Can
             }
         }
 
-        private Task ExecuteSyntaxTreeActions(CancellationToken cancellationToken)
+        public void StartCompleteAnalysis(ImmutableHashSet<SyntaxTree> syntaxTreesToSkip, CancellationToken cancellationToken)
+        {
+            if (_initializeTaskStarted && _primaryTask == null)
+            {
+                _primaryTask = Task.Run(async () =>
+                    {
+                        await _initializeTask.ConfigureAwait(false);
+                        await ProcessCompilationEventsAsync(true, cancellationToken).ConfigureAwait(false);
+                        await ExecuteSyntaxTreeActions(_compilation.SyntaxTrees.Where((tree) => !syntaxTreesToSkip.Contains(tree)), cancellationToken).ConfigureAwait(false);
```

Comment:
Due to the subtlety of compilation start actions being run before processing any compilation events, swapping this line with the line above actually achieves a valid action ordering. Running syntax tree actions before most other actions, while not required, is also likely to match expectations.